src/capture.cpp

```diff
@@ -2239,7 +2036,15 @@ const char *locale = DEFAULT_LOCALE;
                 }
                 else
                 {
-                    Log(2, "Using the last night exposure of %s\n", length_in_units(current_exposure_us, true));
+					// If gain changes, we have to change the exposure time to get an equally
+					// exposed image.
+					// ZWO gain has unit 0.1dB, so we have to convert the gain values to a factor first
+					//    newExp = (oldExp * oldGain) / newGain
+					// e.g.  20s = (10s    * 2.0)     / (1.0) 
+
+                    // current values are here the last night values
+                    current_exposure_us = (current_exposure_us * pow(10, (float)currentGain / 10.0 / 20.0)) / pow(10, (float)asiDayGain / 10.0 / 20.0);
```

Comment:
make a fuction - this would be testable
https://github.com/AndreasLMeg/allsky/blob/li_overlay_text/src/allsky_helper.cpp
https://github.com/AndreasLMeg/allsky/blob/li_overlay_text/test/unittest/UT_Allsky.cpp

Reply:
@AndreasLMeg exactly what should be a function?  This?
~~~
current_exposure_us = (current_exposure_us * pow(10, (float)currentGain / 10.0 / 20.0)) / pow(10, (float)asiDayGain / 10.0 / 20.0);
~~~

Should I do that now, or can it wait until after this PR is merged (which I prefer to do)?

Can't the `gain2zwoGain()` and `zwoGain2gain()` functions go in allsky_common.cpp so we don't need another .cpp and .h file?  Or am I not understanding something?

I like what you are doing with `https://github.com/AndreasLMeg/allsky/blob/li_overlay_text/test/unittest/UT_Allsky.cpp`.  GOOD STUFF!